Dockerfile

```diff
@@ -1,50 +1,119 @@
+
 FROM amazonlinux:2
 
 # Set up working directories
-RUN mkdir -p /opt/app
-RUN mkdir -p /opt/app/build
-RUN mkdir -p /opt/app/bin/
-
-# Copy in the lambda source
-WORKDIR /opt/app
-COPY ./*.py /opt/app/
-COPY requirements.txt /opt/app/requirements.txt
+RUN mkdir -p \
+    /opt/app \
+    /opt/app/build \
+    /opt/app/bin \
+    /opt/app/python_deps \
+    /opt/app/cli
 
 # Install packages
-RUN yum update -y
-RUN yum install -y cpio python3-pip yum-utils zip unzip less
-RUN yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
-
-# This had --no-cache-dir, tracing through multiple tickets led to a problem in wheel
-RUN pip3 install -r requirements.txt
-RUN rm -rf /root/.cache/pip
+RUN yum update -y \
+    && amazon-linux-extras install epel -y \
+    && yum install -y \
+      cpio \
+      yum-utils \
+      tar.x86_64 \
+      gzip \
+      zip \
+      python3-pip \
+      shadow-utils.x86_64 \
+    && yum clean all \
+    && rm -rf /var/cache/yum
 
 # Download libraries we need to run in lambda
 WORKDIR /tmp
-RUN yumdownloader -x \*i686 --archlist=x86_64 clamav clamav-lib clamav-update json-c pcre2 libprelude gnutls libtasn1 lib64nettle nettle
-RUN rpm2cpio clamav-0*.rpm | cpio -idmv
-RUN rpm2cpio clamav-lib*.rpm | cpio -idmv
-RUN rpm2cpio clamav-update*.rpm | cpio -idmv
-RUN rpm2cpio json-c*.rpm | cpio -idmv
-RUN rpm2cpio pcre*.rpm | cpio -idmv
-RUN rpm2cpio gnutls* | cpio -idmv
-RUN rpm2cpio nettle* | cpio -idmv
-RUN rpm2cpio lib* | cpio -idmv
-RUN rpm2cpio *.rpm | cpio -idmv
-RUN rpm2cpio libtasn1* | cpio -idmv
-
-# Copy over the binaries and libraries
-RUN cp /tmp/usr/bin/clamscan /tmp/usr/bin/freshclam /tmp/usr/lib64/* /opt/app/bin/
+RUN yumdownloader -x \*i686 --archlist=x86_64 \
+      clamav \
+      clamav-lib \
+      clamav-update \
+      clamav-scanner-systemd \
+      elfutils-libs \
+      json-c \
+      lz4 \
+      pcre2 \
+      systemd-libs \
+      libtool-ltdl \
+      libxml2 \
+      bzip2-libs \
+      xz-libs \
+      libprelude \
+      gnutls \
+      nettle \
+    && rpm2cpio clamav-0*.rpm | cpio -vimd \
+    && rpm2cpio clamav-lib*.rpm | cpio -vimd \
+    && rpm2cpio clamav-update*.rpm | cpio -vimd \
+    && rpm2cpio json-c*.rpm | cpio -vimd \
+    && rpm2cpio pcre*.rpm | cpio -vimd \
+    && rpm2cpio libtool-ltdl*.rpm | cpio -vimd \
+    && rpm2cpio libxml2*.rpm | cpio -vimd \
+    && rpm2cpio bzip2-libs*.rpm | cpio -vimd \
+    && rpm2cpio xz-libs*.rpm | cpio -vimd \
+    && rpm2cpio libprelude*.rpm | cpio -vimd \
+    && rpm2cpio gnutls*.rpm | cpio -vimd \
+    && rpm2cpio nettle*.rpm | cpio -vimd \
+    && rpm2cpio clamd-0*.rpm | cpio -idmv \
+    && rpm2cpio elfutils-libs*.rpm | cpio -idmv \
+    && rpm2cpio lz4*.rpm | cpio -idmv \
+    && rpm2cpio systemd-libs*.rpm | cpio -idmv \
+    && cp -r \
+      /tmp/usr/bin/clamdscan \
+      /tmp/usr/sbin/clamd \
+      /tmp/usr/bin/freshclam \
+      /tmp/usr/lib64/* \
+      /usr/lib64/libpcre.so* \
```

Comment:
must copy all lipcre.so variants using wildcard, because lipcre.so.1 is a symbolic link sometimes